src/main/java/build/buildfarm/worker/shard/ShardWorkerContext.java

```diff
@@ -182,12 +194,14 @@
     this.limitExecution = limitExecution;
     this.limitGlobalExecution = limitGlobalExecution;
     this.onlyMulticoreTests = onlyMulticoreTests;
+    this.keepInsertFiles = keepInsertFiles;
     Preconditions.checkState(
         !limitGlobalExecution || limitExecution,
         "limit_global_execution is meaningless without limit_execution");
     Preconditions.checkState(
         !onlyMulticoreTests || limitExecution,
         "only_multicore_tests is meaningless without limit_execution");
+    workerStubs = WorkerStubs.create(getDigestUtil());
```

Comment:
You don't want to create a separate workerStubs cache here - the worker already has one for doing remote cas downloads.